src/org/labkey/test/components/ui/search/SampleFinder.java

```diff
@@ -0,0 +1,221 @@
+                .findElements(elementCache().filterCardsSection);
+        Collections.reverse(removeButtons);
+        // Don't wait for search results to update after each removed card
+        for (WebElement button : removeButtons)
+        {
+            button.click();
+            getWrapper().shortWait().until(ExpectedConditions.stalenessOf(button));
+        }
+        getWrapper().shortWait().withMessage("Clearing all search cards").until(wd -> isEmptySearch());
+
+        clearElementCache();
+        return this;
+    }
+
+    /**
+     * Waiter that will wait for the search results to load. This is a hook for perf tests to support longer waits.
+     * @return WebDriverWait to be used by {@link #waitForReady()}
+     */
+    protected WebDriverWait loadingWait()
+    {
+        return getWrapper().shortWait();
+    }
+
+    private boolean isEmptySearch()
+    {
+        return Locator.css(".filter-cards.empty").isDisplayed(this);
+    }
+
+    private void doAndWaitForUpdate(Runnable func)
+    {
+        WebElement resultsGridEl = elementCache().resultsGrid.getComponentElement();
+        if (resultsGridEl.isDisplayed())
+        {
+            func.run();
+            getWrapper().shortWait().until(ExpectedConditions.stalenessOf(resultsGridEl));
+        }
+        else
+        {
+            func.run();
+        }
+
+        clearElementCache();
+        elementCache(); // waitForReady
+    }
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+    protected class ElementCache extends Component<?>.ElementCache
+    {
+        final WebElement panelHeading = Locator.byClass("panel-heading").findWhenNeeded(this);
+
+        WebElement findAddParentButton(String parentNoun)
+        {
+            return BootstrapLocators.button(parentNoun + " Properties")
+                    .withChild(Locator.byClass("container--addition-icon")).findElement(panelHeading);
+        }
+
+        final WebElement filterCardsSection = Locator.byClass("filter-cards").findWhenNeeded(this);
+        List<FilterCard> findFilterCards()
+        {
+            return Locator.byClass("filter-cards__card").findElements(filterCardsSection)
+                    .stream().map(FilterCard::new).collect(Collectors.toList());
+        }
+
+        FilterCard findFilterCard(String queryName)
+        {
+            return new FilterCard(Locator.byClass("filter-cards__card").withDescendant(
+                    Locator.byClass("primary-text").withText(queryName)).findElement(filterCardsSection));
+        }
+
+        final TabbedGridPanel resultsGrid = new TabbedGridPanel.TabbedGridPanelFinder(getDriver()).findWhenNeeded(this);
+    }
+
+    /**
+     * Represents a single filter card. Don't expose outside this class because its lifecycle is confusing
+     */
+    private class FilterCard extends Component<Component<?>.ElementCache>
```

Comment:
You should add a getter that returns the filters that have been applied in the given card:

<img width="343" alt="Screen Shot 2022-03-14 at 5 37 33 PM" src="https://user-images.githubusercontent.com/12738200/158283504-e8058f3a-d7f1-402f-b995-7b719edcd60a.png">